csharp/src/Apache.Arrow/Types/NestedType.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Apache.Arrow.Types
+{
+    public abstract class NestedType : ArrowType
+    {
+        public IList<Field> Children { get; }
+
+        public Field Child(int i) => Children[i];
```

Comment:
Why have this method when callers can just use the `Children` field to index into the collection? I don't think it provides much value over the `IList<Field> Children` property.

Reply:
I added this method in order to prevent rewriting of a child. ( Like the [C++ implementation](https://github.com/apache/arrow/blob/master/cpp/src/arrow/type.h#L229))

On second thought,  it doesn't make much sense. I think it is better to change the type of the Children property to IReadOnlyList<Field> and remove the Child method.